### PR TITLE
items: add temporary location name by serializer

### PR DESCRIPTION
**Pull request by Garfield-fr**
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

File: rero_ils/modules/items/serializers/json.py [hunk 1]
```diff
@@ -119,8 +128,3 @@ def post_process_serialize_search(self, results, pid_fetcher):
             filter_document_type_buckets(buckets)
 
         return super().post_process_serialize_search(results, pid_fetcher)
-
-        # Correct document type buckets
```
Comment (rerowep) [inline rero_ils/modules/items/serializers/json.py:123]: If you delete this the facette for the documents are not displayed correctly.

**Review by rerowep — CHANGES_REQUESTED** [review]

File: rero_ils/modules/items/serializers/json.py [hunk 2]
```diff
@@ -71,6 +71,15 @@ def post_process_serialize_search(self, results, pid_fetcher):
             collection = item.in_collection()
             if collection:
                 metadata['in_collection'] = collection
+
+            # Temporary location
+            if 'temporary_location' in metadata:
+                temp_location = metadata['temporary_location'];
```
Comment (rerowep) [inline rero_ils/modules/items/serializers/json.py:77]: Better to write:
```
temp_location = metadata['temporary_location']
    if temp_location:
```

File: rero_ils/modules/items/serializers/json.py (hunk 1 above)
Comment (Garfield-fr) [reply rero_ils/modules/items/serializers/json.py:123]: this code was after the return. It is also before the return. Code cleaning.

File: rero_ils/modules/items/serializers/json.py [hunk 3]
```diff
@@ -71,6 +71,15 @@ def post_process_serialize_search(self, results, pid_fetcher):
...
+                if temp_location['pid'] not in locs:
+                    locs[temp_location['pid']] = Location \
+                        .get_record_by_pid(temp_location['pid'])
+                temp_location['name'] = locs[temp_location['pid']].get('name')
+
```
Comment (rerowep) [inline rero_ils/modules/items/serializers/json.py:82]:
```
temp_location_pid = temp_location['pid']
if temp_location_pid not in locs:
    locs[temp_location_pid] = Location.get_record_by_pid(temp_location_pid)
temp_location['name'] =  locs[temp_location_pid].get('name')
```

File: rero_ils/modules/items/serializers/json.py (hunk 1 above)
Comment (rerowep) [reply rero_ils/modules/items/serializers/json.py:123]: You are right this code exist also in line 116.

**Review by rerowep — APPROVED** [review]

**Pull request merged**